patches/minecraft/net/minecraft/world/level/block/RedStoneWireBlock.java.patch

```diff
@@ -0,0 +1,39 @@
+--- a/net/minecraft/world/level/block/RedStoneWireBlock.java
++++ b/net/minecraft/world/level/block/RedStoneWireBlock.java
+@@ -199,7 +_,8 @@
+       BlockState blockstate = p_55523_.m_8055_(blockpos);
+       if (p_55526_) {
+          boolean flag = this.m_55612_(p_55523_, blockpos, blockstate);
+-         if (flag && m_55640_(p_55523_.m_8055_(blockpos.m_7494_()))) {
++         flag &= p_55523_.m_8055_(blockpos.m_7494_()).canRedstoneConnectTo(p_55523_, blockpos.m_7494_(), null);
++         if (flag) {
+             if (blockstate.m_60783_(p_55523_, blockpos, p_55525_.m_122424_())) {
+                return RedstoneSide.UP;
+             }
+@@ -207,8 +_,9 @@
+             return RedstoneSide.SIDE;
+          }
+       }
+-
+-      return !m_55594_(blockstate, p_55525_) && (blockstate.m_60796_(p_55523_, blockpos) || !m_55640_(p_55523_.m_8055_(blockpos.m_7495_()))) ? RedstoneSide.NONE : RedstoneSide.SIDE;
++      boolean flag1 = blockstate.canRedstoneConnectTo(p_55523_, blockpos, p_55525_);
++      boolean flag2 = p_55523_.m_8055_(blockpos.m_7495_()).canRedstoneConnectTo(p_55523_, blockpos.m_7495_(), null);
++      return !flag1 && (blockstate.m_60796_(p_55523_, blockpos) || !flag2) ? RedstoneSide.NONE : RedstoneSide.SIDE;
```

Comment:
You store the output of `canRedstoneConnectTo` for the blockpos below in a `flag2` variable, which means any expensive operations could potentially be calculated for no reason if the boolean condition can be exited early because either `flag1` is true or `isRedstoneConductor`. I would also stray away from choosing variable names that forgeflower will automatically choose, as you've done here. This means if another boolean variable were to be added to the method, it would conflict with the one you declare. It's always better to choose a readable variable name over 2 variables only differing by a number, especially when that conflicts with what the decompiler generates for local variable names.

Reply:
In the latest commit, that long line of return has been replaced with if statements, making it more readable. Unnecessary function calls are avoided.